src/variables/utils/validation.ts

```diff
@@ -17,8 +17,24 @@ export const validateVariableName = (
   varName: string,
   id?: string
 ): {error: string | null} => {
-  if ((varName || '').match(/^\s*$/)) {
-    return {error: 'Variable name cannot be empty'}
+  const spaceRegex = /^\s*$/
+
+  // must start with a letter or underscore, can only contain letters, numbers, and underscores only
+  const validCharacters = /^[A-Za-z_]+\w*$/
+
+  const emptyErrorText = 'Variable name cannot be empty'
+  const fullErrorText =
+    'Variable name must start with a letter or underscore, and ' +
+    'contain only numbers, letters, and underscores.'
```

Comment:
It looks like you are using these in the unit tests. Why not just export these strings, then? In case there was any need to tweak them, it would be easier just to change the exported value rather than hunt down and update all matching literals of these errors.